packages/respect-core/src/utils/digest-auth/parse-www-authenticate-header.ts

```diff
@@ -0,0 +1,27 @@
+export function parseWwwAuthenticateHeader(wwwAuthenticateHeader: string): {
+  realm?: string;
+  nonce?: string;
+  opaque?: string;
+  qop?: string;
+  algorithm?: string;
+  cnonce?: string;
+  nc?: string;
+} {
+  const headerParts = wwwAuthenticateHeader
+    .replace('Digest ', '')
+    .split(',')
+    .map((part: string) => part.trim());
+
+  const keys = ['realm', 'nonce', 'opaque', 'qop', 'algorithm', 'cnonce', 'nc'];
```

Comment:
Could you add an explanation or a link to docs that explain what is going on here? Looks like some spell 🪄